junit-platform-engine/src/main/java/org/junit/platform/engine/discovery/NestedMethodSelector.java

```diff
@@ -239,4 +243,41 @@ public String toString() {
 				.toString();
 	}
 
+	@Override
+	public Optional<String> toSelectorString() {
+		return nestedClassSelector.toSelectorString().map(parent -> {
+			StringBuilder sb = new StringBuilder(parent)
+					// Not totally happy with how we have to change the prefix here.
+					// Alternativly, we could duplicate the logic of the NestedClassSelector
```

Comment:
Just an idea, you could consider making the scheme recursive, i.e:

`method:nested-class:com.example.Example#!/test(String,String)`.

It works for Jar URIs.

Reply:
I've since reworked this part which resolved the issue.